controllers/operandrequest/reconcile_operand.go

```diff
@@ -1346,7 +1396,8 @@ func (r *Reconciler) deleteK8sResource(ctx context.Context, existingK8sRes unstr
 	if apierrors.IsNotFound(err) {
 		klog.V(3).Infof("There is no k8s resource: %s from kind: %s", name, kind)
 	} else {
-		if r.CheckLabel(k8sResShouldBeDeleted, map[string]string{constant.OpreqLabel: "true"}) && !r.CheckLabel(k8sResShouldBeDeleted, map[string]string{constant.NotUninstallLabel: "true"}) {
+
+		if (newLabels == nil || newLabels[constant.OpreqLabel] == "true") && !r.CheckLabel(k8sResShouldBeDeleted, map[string]string{constant.NotUninstallLabel: "true"}) {
```

Comment:
I am thinking if the condition should be:

The resource would be uninstalled when following conditions are met:
If (A && B || C) && D
   - A: newLabels not empty
   - B: newLabels contains opreq label
   - C: k8sResShouldBeDeleted contains opreq label)
   - D: k8sResShouldBeDeleted NOT contains notUninstall label

```suggestion
		if (newLabels != nil && newLabels[constant.OpreqLabel] == "true" || r.CheckLabel(k8sResShouldBeDeleted, map[string]string{constant.OpreqLabel: "true"})) && !r.CheckLabel(k8sResShouldBeDeleted, map[string]string{constant.NotUninstallLabel: "true"}) {
```